cirq/vis/state_histogram.py

```diff
@@ -50,32 +50,54 @@ def get_state_histogram(result: 'result.Result') -> np.ndarray:
 
 
 def plot_state_histogram(
-    values: Union['result.Result', np.ndarray], ax: Optional['plt.Axis'] = None
+    data: Union['result.Result', collections.Counter, Sequence[SupportsFloat]],
+    ax: Optional['plt.Axis'] = None,
+    *,
+    tick_label: Optional[Sequence[str]] = None,
+    xlabel: Optional[str] = 'qubit state',
+    ylabel: Optional[str] = 'result count',
+    title: Optional[str] = 'Result State Histogram',
```

Comment:
Do we want to allow `None` for these fields? If not, these shouldn't have type `Optional`.

Reply:
Yes, these fields can potentially be none. For eg, we might want to omit the title when drawing a figure to be included in a pdf. 
We also follow the same pattern for other visualization utilities, for eg: `integrated_histogram` 
https://github.com/quantumlib/Cirq/blob/bee1a448bc98b52ab952ccf284c0382918abaad1/cirq/vis/histogram.py#L32